src/test/java/com/gr4vy/sdk/Gr4vyClientTest.java

```diff
@@ -28,6 +29,22 @@ public void getEmbedTokenTest() throws Gr4vyException, ApiException {
     	String token = client.getEmbedToken(embed);
         assert token != null;
     }
+
+	@Test
+    public void getEmbedTokenTestWithCheckoutSessionPassedIn() throws Gr4vyException, ApiException {
+    	Gr4vyClient client = new Gr4vyClient("spider", "private_key.pem");
+		
+		Map<String, Object> embed = new HashMap<String, Object>();
+		embed.put("amount", 1299);
+		embed.put("currency", "USD");
+		
+		CheckoutSessionsApi checkoutSessionsApi = new CheckoutSessionsApi(client.getClient());
+		CheckoutSession checkoutSession = checkoutSessionsApi.newCheckoutSession();
+		embed.put("checkout_session_id", checkoutSession.getId());
+
+    	String token = client.getEmbedToken(embed);
```

Comment:
@RonanFabricante mostly looks good but here we should pass the checkoutSessionId into getEmbedToken (rather than including it in the embed parameters)

Reply:
Thanks @steve-gr4vy, fixed this.